planner/core/resolve_indices.go

```diff
@@ -632,13 +635,60 @@ func (p *baseSchemaProducer) ResolveIndices() (err error) {
 	return
 }
 
+// ResolveIndices implements Plan interface.
+func (p *basePhysicalJoin) ResolveIndices() (err error) {
+	for _, child := range p.children {
+		err = child.ResolveIndices()
+		if err != nil {
+			return err
+		}
+	}
+	switch p.JoinType {
+	case RightOuterJoin:
+		p.AlignNullFlag(p.children[1].Schema())
+		p.EnsureNullFlag(p.schema, 0, len(p.children[0].Schema().Columns))
+	case LeftOuterJoin:
+		p.AlignNullFlag(p.children[0].Schema())
+		p.EnsureNullFlag(p.schema, len(p.children[0].Schema().Columns), len(p.schema.Columns))
+	default:
+		p.AlignNullFlag(p.children[0].Schema())
+		p.AlignNullFlag(p.children[1].Schema())
+	}
+	return
+}
+
 // ResolveIndices implements Plan interface.
 func (p *basePhysicalPlan) ResolveIndices() (err error) {
 	for _, child := range p.children {
 		err = child.ResolveIndices()
 		if err != nil {
 			return err
 		}
+		p.AlignNullFlag(child.Schema())
```

Comment:
The non-join plan that has multi-children is "Union" and "Apply". As I know, at least for union, the nullable should be true when at least one corresponding column is nullable.
These plans have none bussiness of TiFlash, I suggest you delete this line of code.

Reply:
Strictly speaking, the NotNull flag of the projected columns in Projection and Aggregation should be updated as well?